### PR TITLE
Remove dependency on Shouldy from Services. Fixes #8, #9

### LINKED ISSUE #8: Nuget: "Services" has a dependency on Shouldly...
...But isn't defined as a dependency via nuget. So you need to install Shouldly separately to get it to work.

Current version: **0.81.0-dev**

### PR DESCRIPTION
**Pull request by ctolkien**
Replaces with our own Guard implementation.

### CODE REVIEW

File: Code/OpenRealEstate.Core/Guard.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+
+namespace OpenRealEstate.Core
+{
+    public class Guard
```
Comment (PureKrome) [inline Code/OpenRealEstate.Core/Guard.cs]: Make this class `internal`.

Comment (PureKrome) [inline Code/OpenRealEstate.Core/Guard.cs]: .. and `static` ....

File: Code/OpenRealEstate.Core/Guard.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        public static void AgainstNulls(object obj)
+        {
+            if (obj == null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+        }
+
+        public static void AgainstNulls(string stringObj)
```
Comment (PureKrome) [inline Code/OpenRealEstate.Core/Guard.cs]: Rename to `AgainstNullsAndWhiteSpace`.

File: Code/OpenRealEstate.Core/Guard.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            if (string.IsNullOrEmpty(stringObj))
```
Comment (PureKrome) [inline Code/OpenRealEstate.Core/Guard.cs]: `IsNullOrWhitspace()`

File: Code/OpenRealEstate.Core/Guard.cs (hunk 2 above)
Comment (PureKrome) [inline Code/OpenRealEstate.Core/Guard.cs]: Check if `WhiteSpace` or `Whitespace` (whatever the method if called on `string`)

File: Code/OpenRealEstate.Core/Guard.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+
+
+namespace OpenRealEstate.Core
+{
+    public class Guard
+    {
+        public static void AgainstNulls(object obj)
```
Comment (PureKrome) [inline Code/OpenRealEstate.Core/Guard.cs]: `AgainstNull` (no `s`)

File: Code/OpenRealEstate.Core/Guard.cs (hunk 2 above)
Comment (PureKrome) [inline Code/OpenRealEstate.Core/Guard.cs]: also, `Null`, not `Nulls`...

File: Code/OpenRealEstate.Core/Guard.cs (hunk 1 above)
Comment (ctolkien) [inline Code/OpenRealEstate.Core/Guard.cs]: Can't make it internal, this is in the `Core` proj, wouldn't not be accessible in `Services` proj  (presumably we'd further roll this out for all `ArgumentNullExceptions` in other projects too, hence why I did not create the Guard class in `Services`)

**Comment by PureKrome** [comment]
Move the class to the `Services` project then.

**Comment by ctolkien** [comment]
Done, as mentioned tho, means we can't now roll it out through the other projects.

**Comment by PureKrome** [comment]
:shipit:

**Comment by PureKrome** [comment]
- Fixes #8 
- Fixes #9

**Pull request merged**